RestaurantReservation/Program.cs

```diff
@@ -1,2 +1,34 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using RestaurantReservation.Db;
+using RestaurantReservation.Db.Repositories;
+using RestaurantReservation.Db.RepositoriesInterfaces;
+
+public class Program
+{
+    public static void Main(string[] args)
+    {
+        var host = CreateHostBuilder(args).Build();
+        host.Run();
+    }
+
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureServices((context, services) =>
+            {
+                services.AddDbContext<RestaurantReservationDbContext>(options =>
+                    options.UseSqlServer(context.Configuration.GetConnectionString("RestaurantReservationCore")));
+
+                services.AddSingleton<DbContextFactory>();
+                services.AddScoped<ICustomerRepository, CustomerRepository>();
```

Comment:
Using DbContextFactory is good if you need to create DbContext instances outside of typical DI scopes.
 
 but in our case you do not need the factory as you can inject DBContext directly 
 
 also since the library here is handling DI and it's not manual 
 you do not need to implement IDisposable yourself. The DbContext is managed by the Dependency Injection (DI) container in ASP.NET Core, which automatically handles its lifecycle.